src/components/Dropdown/Dropdown.tsx

```diff
@@ -179,13 +198,31 @@ export class Dropdown extends Component<DropdownProps> {
     };
 
     const passDropdownItemProps = {
-      ...dropdownItemProps,
-      className: classnames(
-        dropdownClassNames.item,
-        dropdownItemProps.className,
-      ),
+      className: classnames(dropdownClassNames.item, dropdownItemClassName),
     };
 
+    const onChange = (newValue: string) => {
+      if (!!propOnChange) {
+        propOnChange(newValue);
+      }
+      if (!('value' in this.props)) {
+        this.setState({ selectedValue: newValue });
+      }
+    };
+
+    const listboxInputProps = {
+      'aria-labelledby': ariaLabelledByIds,
+      onChange,
+      name,
+      value: selectedValue || '',
+    };
+
+    const listboxDisplayValue = alwaysShowVisualPlaceholder
+      ? visualPlaceholder
+      : !!selectedValue
+      ? ''
+      : visualPlaceholder;
```

Comment:
If alwaysShowVisualPlaceholder is false and selectedValue has a value, this seems to return an empty string. Why is that?

Reply:
Reach handles empty string as no value and uses the selected item node as button display value. We still want to explicitly set the prop to make it work properly => if always show is set, we replace the button value with the placeholder alway, if it's not, we display the selected node (can be string or react element) and if nothing is selected, we show the placeholder.